### PR TITLE
feat(plugins/rate-limiting): Add ability to rate limit by subdomain

### PR DESCRIPTION
**Pull request by theundeadmonk**
### Summary
Kong provides a basic rate limiting plugin. This plugin however does not offer the ability to rate limit by subdomain. Something which is used by a lot of organizations to achieve multi tenancy.
This PR adds support to the rate limiting plugin to rate limit by subdomain.

### Checklist

- [x] The Pull Request has tests
- [ ] There's an entry in the CHANGELOG
- [x] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - https://github.com/Kong/docs.konghq.com/pull/4962

### Full changelog

* [Implement ...]

### Issue reference

Fix #_[issue number]_

### CODE REVIEW

**Comment by chronolaw** [comment]
Thank you, [theundeadmonk](https://github.com/theundeadmonk), I think that we should add test cases for this feature.

**Comment by theundeadmonk** [comment]
@chronolaw Do you mind taking a look at this again?

**Comment by vm-001** [comment]
@theundeadmonk Good idea! I was thinking of another alternative solution. Have you ever tried the limit_by header feature? which is basically setting the `config.limit_by` to `header` and writing a custom plugin to extract the subdomain and add it as a request header. I think this is a more general approach to implementing a dynamic limit_by. 

And of cause, adding an extra built-in limit_by is something convenient for users. (Let's seek some opinions from other members to evaluate this PR)

(By the way, can you prepare a corresponding docs PR on https://github.com/Kong/docs.konghq.com, please? https://github.com/Kong/docs.konghq.com/pull/4626 is an example in case you need)

**Comment by theundeadmonk** [comment]
@vm-001 Added a PR to the docs repo if you still feel this PR might be useful
https://github.com/Kong/docs.konghq.com/pull/4962

**Comment by ADD-SP** [comment]
@hbagdi What do you think?

File: kong/plugins/rate-limiting/handler.lua [hunk 1]
```diff
@@ -74,6 +74,11 @@ local function get_identifier(conf)
     if req_path == conf.path then
       identifier = req_path
     end
+
+  elseif conf.limit_by == "subdomain" then
+    local host = kong.request.get_host()
+    local subdomain = host:match("(.+)%.")
```
Comment (StarlightIbuki) [inline kong/plugins/rate-limiting/handler.lua:80]: Do we need a more robust way to extract subdomains? Some local hostnames are without a dot; some local domains are ended with a dot.

**Review by StarlightIbuki — COMMENTED** [review]
Could you explain the scenario?
At first glance, it could be solved with route/service-scoped RL plugins.
And is there a reason to limit by subdomain, instead of the domain(the host)? I don't feel the necessity to have `a.b` and `a.c` share the same namespace, and the host should be able to tell the difference between `a.b` and `c.b`.

**Comment by hbagdi** [comment]
Deferring the decision to @teb510

**Comment by hbagdi** [comment]
Closing this due to lack of activity. Please re-open if needed.

**Pull request closed without merging**